Week1 [Step 2] juho

JuhoJUNG-Kr opened the pull request:
@Kioding 
step 2 입니다. 반복문 안에 함수를 넣어 출력하였습니다. 함수 안에 반복문을 넣어 출력했을 시, 결과값이 이상하게 도출되는 것을 확인했습니다.
<img width="1176" alt="스크린샷 2022-02-09 15 45 12" src="https://user-images.githubusercontent.com/94790678/153372398-4dd496aa-a259-4899-b94e-28f53ba283f0.png">
혹시 왜 이렇게 작동하는 지 알수 있을까요?

JuhoJUNG-Kr committed:
first commit for step2

JuhoJUNG-Kr committed:
second commit

JuhoJUNG-Kr committed:
third commit

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,23 @@
 
 import Foundation
 
-print("Hello, World!")
```
불필요한 코드의 삭제 좋습니다!

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,23 @@
+        
+        print("   |   | ")
+    }
+}
```
- bodyOfIceCream() 의 for문 
`for _ in 0...7{`
- tailOfIcecream() 의 for 문
`for _ in 0...3 {`

서로 컨벤션이 다른데, 통일성을 갖춰보면 좋을 것 같습니다 :)

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,23 @@
-print("Hello, World!")
 
+func bodyOfIcecream() {
+    
```
함수 사이에 한칸 씩 띄어진 경우가 많은데 혹시 이유가 있는지 궁금하네요 !

JuhoJUNG-Kr replied:
제가 보기에 넓찍한게 좋아보여서 하다보니 습관이 된 것 같습니다. 조속히 고치도록 할게요!

JuhoJUNG-Kr committed:
Final

JuhoJUNG-Kr committed:
Test for commit template
style: change convention for func
docs: change loadmap for code
커밋 템플릿 테스트용입니다.

-

JuhoJUNG-Kr reviewed (commented):
드디어 됬당 !

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,17 @@
-print("Hello, World!")
 
+func bodyOfIcecream(){
+    for _ in 0...7{
```
숫자 뒤에 `{` 는 한칸 띄는 게 더 좋아보이네요 :)

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,17 @@
+    for _ in 0...3{
+        print("  |   | ")
+    }
+}
```
직접적인 값으로 `0...3` 을 넣어주셨는데 처음 이 코드를 본다면 무슨 뜻인지 이해하기 어려울 것 같아요.
상수를 이용하여 나타내어보면 좋을 것 같아요!

`print("  |   | ")`
이 부분도 무엇을 print하려는 걸까요?! 위와 같은 방법으로 바꿔보면 좋겠습니다.

이 코멘트는 `bodyOfIcecream()` 에도 해당합니다  😆

JuhoJUNG-Kr replied:
각각 수정하였습니다!

JuhoJUNG-Kr committed:
Real_Final_commit for step2
style: 코드 정렬
docs: 순서도 수정

 1. 함수의 기능을 확실히 알 수 있도록 함수명 변경
 2. 함수명 변경의 따른 순서도 업데이트

-

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,18 @@
+    for bodyCount in 0...bodyCount {
+        print("***********")
+    }
+}
```
`bodyCount` 를 for문에서 선언하였는데 실제로는 사용하고 있지 않네요?!

Kioding replied:
`***********` 이것도 지속적으로 사용한다면 상수로 해줄 수도 있을 것 같아요 :)

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,18 @@
+    for tailCount in 0...tailCount {
+        print("  |   | ")
+    }
+}
```
위와 같은 코멘트입니다 :)

JuhoJUNG-Kr committed:
Step2 change for code
style: change for code form
docs: update
1.아이스크림의 몸통을 호출하는 함수안에 반복문을 두번 호출하였습니다.
순서도 안에 코드를 삭제하고 문장으로 표현하였습니다.
2.반복문을 두 번 호출하여 아이스크림 몸통의 내부 구조가 바뀔시
더 편하게 수정할 수 있도록 하기위해 수정했습니다.

-

JuhoJUNG-Kr committed:
Refactor:change document for code flow

-

JuhoJUNG-Kr committed:
Refactor: change for step2 code and doucument

-

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -10,17 +10,19 @@ import Foundation
     for _ in 0...bodyLonger {
         for _ in 0...bodyWidth {
-            print("*", terminator:"")
+            print(bodyOfIcecream, terminator:"")
```
상수로 넣어주시니 어떤 걸 print 하는지 보기 좋네요!
